File: sample/src/main/java/me/xiaopan/sketchsample/adapter/itemfactory/PageMenuItemFactory.java
package me.xiaopan.sketchsample.adapter.itemfactory;

import android.content.Context;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.OnClick;
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory;
import me.xiaopan.sketchsample.R;
import me.xiaopan.sketchsample.activity.MainActivity;
import me.xiaopan.sketchsample.adapter.BindAssemblyRecyclerItem;

public class PageMenuItemFactory extends AssemblyRecyclerItemFactory<PageMenuItemFactory.PageMenuItem> {
    private OnClickItemListener onClickItemListener;

    public PageMenuItemFactory(OnClickItemListener onClickItemListener) {
        this.onClickItemListener = onClickItemListener;
    }

    @Override
    public boolean isTarget(Object o) {
        return o instanceof MainActivity.Page;
    }

    @Override
    public PageMenuItem createAssemblyItem(ViewGroup viewGroup) {
        return new PageMenuItem(R.layout.list_item_page_menu, viewGroup);
    }

    public interface OnClickItemListener {
        void onClickItem(MainActivity.Page page);
    }

    public class PageMenuItem extends BindAssemblyRecyclerItem<MainActivity.Page> {
        @BindView(R.id.text_pageMenuItem)
        TextView textView;

        public PageMenuItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @OnClick(R.id.text_pageMenuItem)
        void clickItem() {
            onClickItemListener.onClickItem(getData());
        }

        @Override
        protected void onConfigViews(Context context) {

        }

        @Override
        protected void onSetData(int i, MainActivity.Page pageMenu) {
            textView.setText(pageMenu.getName());
        }
    }
}

File: sample/src/main/java/me/xiaopan/sketchsample/adapter/itemfactory/PageMenuItemFactory.kt
package me.xiaopan.sketchsample.adapter.itemfactory

import android.content.Context
import android.view.ViewGroup
import android.widget.TextView
import me.xiaopan.assemblyadapter.AssemblyRecyclerItem
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory
import me.xiaopan.sketchsample.R
import me.xiaopan.sketchsample.activity.MainActivity
import me.xiaopan.ssvt.bindView

class PageMenuItemFactory(private val onClickItemListener: OnClickItemListener) : AssemblyRecyclerItemFactory<PageMenuItemFactory.PageMenuItem>() {

    override fun isTarget(o: Any): Boolean {
        return o is MainActivity.Page
    }

    override fun createAssemblyItem(viewGroup: ViewGroup): PageMenuItem {
        return PageMenuItem(R.layout.list_item_page_menu, viewGroup)
    }

    interface OnClickItemListener {
        fun onClickItem(page: MainActivity.Page)
    }

    inner class PageMenuItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyRecyclerItem<MainActivity.Page>(itemLayoutId, parent) {
        val textView: TextView by bindView(R.id.text_pageMenuItem)

        override fun onConfigViews(context: Context) {
            textView.setOnClickListener { onClickItemListener.onClickItem(data) }
        }

        override fun onSetData(i: Int, pageMenu: MainActivity.Page) {
            textView.text = pageMenu.showName
        }
    }
}
